core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -906,6 +908,25 @@ func (o *evmTxStore) FindReceiptsPendingConfirmation(ctx context.Context, blockN
 	return
 }
 
+// FindTxWithRequestID returns any broadcast ethtx with the given requestID
+func (o *evmTxStore) FindTxWithRequestID(requestID common.Hash) (etx *Tx, err error) {
```

Comment:
Instead of this function, you could directly add this logic inside CreateTransaction.
I see CreateTransaction already has this logic, for the field txRequest.PipelineTaskRunID.

It might be easier to just repeat that logic as it is, for your new requestId field.

Reply:
I was considering that originally but thought it'd be better to maintain this logic in the TXM `CreateTransaction` so it's generalized. Otherwise when creating a new TxStore, the developer would be expected to include this logic. If that's not something we don't need to worry about, I can move this there.